### ISSUE TITLE
Generic type factory: property does not exist on type (TS 3.5 issue)

### ISSUE THREAD

**Issue by sisp** [mobxjs/mobx-state-tree#1343]
**_Bug report_**

* [x] I've checked documentation and searched for existing issues
* [x] I've made sure my project is based on the latest MST version

**Sandbox link or minimal reproduction code**
The [sandbox](https://codesandbox.io/s/friendly-wind-c2xw2) does not show the error because I cannot select TS 3.5. Copy the content of `index.ts` into a TS 3.5 environment instead, e.g. VSCode.

**Describe the expected behavior**
`self.a.someView` should be `null` without errors.

**Describe the observed behavior**
The following error appears when hovering `self.a.someView` in `index.ts` line 18:

```
Property 'someView' does not exist on type 'STNValue<ModelInstanceType<ModelPropertiesDeclarationToProperties<T>, { readonly someView: null; }>, IModelType<ModelPropertiesDeclarationToProperties<T>, { readonly someView: null; }, _NotCustomized, _NotCustomized>>'.
  Property 'someView' does not exist on type 'Exclude<ModelInstanceType<ModelPropertiesDeclarationToProperties<T>, { readonly someView: null; }>, object>'.ts(2339)
```

The error only appears with TS 3.5. TS 3.4 works fine.

**Comment by mweststrate** [comment]
Seems to be fixed by #1346 as well

**Comment by mweststrate** [comment]
Should be solved in 3.14.1
